Golem/Golem.cs

```diff
@@ -133,21 +154,30 @@ public Task<bool> Suspend()
             throw new NotImplementedException();
         }
 
-        public Golem(string golemPath, string? dataDir)
+        public Golem(string golemPath, string? dataDir, ILoggerFactory? loggerFactory = null)
         {
-            var prov_datadir = dataDir != null ? Path.Combine(dataDir, "provider") : null;
-            var yagna_datadir = dataDir != null ? Path.Combine(dataDir, "yagna") : null;
-
-            Yagna = new YagnaService(golemPath, yagna_datadir);
-            Provider = new Provider(golemPath, prov_datadir);
+            var prov_datadir = Path.Combine(dataDir, "provider");
+            var yagna_datadir = Path.Combine(dataDir, "yagna");
+            loggerFactory = loggerFactory == null ? NullLoggerFactory.Instance : loggerFactory;
```

Comment:
This could be changed so that the default value of the param is `NullLoggerFactory.Instance`

Reply:
```
Default parameter value for 'loggerFactory' must be a compile-time constant[CS1736](https://msdn.microsoft.com/query/roslyn.query?appId%3Droslyn%26k%3Dk%28CS1736%29)
(field) static readonly NullLoggerFactory NullLoggerFactory.Instance
Returns the shared instance of NullLoggerFactory.
'Instance' is not null here.
```
🤷